### PR TITLE
RequestTypeMatch predicate

### PR DESCRIPTION
**Pull request by jyotimahapatra**
This PR adds the [RequestTypeMatch](https://github.com/envoyproxy/xds-relay/blob/master/api/protos/aggregation/v1/aggregation.proto#L50-L54) predicate

### CODE REVIEW

**Comment by jyotimahapatra** [comment]
PTAL @jessicayuen @LisaLudique @eapolinario

**Comment by jyotimahapatra** [comment]
PTAL @jessicayuen @LisaLudique @eapolinario

File: internal/app/mapper/mapper_test.go [hunk 1]
```diff
@@ -34,7 +35,25 @@ var postivetests = []TableEntry{
 			getAnyMatch(true),
 			getResultStringFragment(),
 			clusterTypeURL,
-			stringfragment,
+			stringFragment,
+		},
+	},
+	{
+		Description: "RequestTypeMatch Matches with a single typeurl",
```
Comment (jessicayuen) [inline internal/app/mapper/mapper_test.go]: Do we need to add a test case for ordering of the type url? (second match vs first)

Comment (jyotimahapatra) [reply internal/app/mapper/mapper_test.go]: good idea. Done

**Pull request merged**